src/Bicep.LangServer/Handlers/BicepDeploymentScopeRequestHandler.cs

```diff
@@ -0,0 +1,147 @@
+            IConfigurationManager configurationManager,
+            IFeatureProvider features,
+            IFileResolver fileResolver,
+            IModuleDispatcher moduleDispatcher,
+            INamespaceProvider namespaceProvider)
+        {
+            this.compilationManager = compilationManager;
+            this.configurationManager = configurationManager;
+            this.emitterSettings = emitterSettings;
+            this.features = features;
+            this.fileResolver = fileResolver;
+            this.moduleDispatcher = moduleDispatcher;
+            this.namespaceProvider = namespaceProvider;
+        }
+
+        public Task<BicepDeploymentScopeResponse> Handle(BicepDeploymentScopeParams request, CancellationToken cancellationToken)
+        {
+            var documentUri = request.TextDocument.Uri;
+
+            Compilation? compilation;
+
+            try
+            {
+                compilation = GetCompilation(documentUri);
+                var deploymentScope = GetDeploymentScope(compilation.GetEntrypointSemanticModel().TargetScope);
+
+                return Task.FromResult(new BicepDeploymentScopeResponse(deploymentScope, GetCompiledFile(compilation, documentUri), null));
+            }
+            catch (Exception exception)
+            {
+                return Task.FromResult(new BicepDeploymentScopeResponse(ResourceScope.None.ToString(), null, exception.Message));
+            }
+        }
+
+        private static string GetDeploymentScope(ResourceScope resourceScope) => resourceScope switch
+        {
+            ResourceScope.ResourceGroup => LanguageConstants.TargetScopeTypeResourceGroup,
+            ResourceScope.Subscription => LanguageConstants.TargetScopeTypeSubscription,
+            ResourceScope.ManagementGroup => LanguageConstants.TargetScopeTypeManagementGroup,
+            ResourceScope.Tenant => LanguageConstants.TargetScopeTypeTenant,
+            _ => resourceScope.ToString()
+        };
+
+        private string GetCompiledFile(Compilation compilation, DocumentUri documentUri)
+        {
+            var fileUri = documentUri.ToUri();
+
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == fileUri);
+
+            if (diagnosticsByFile.Value.Any(x => x.Level == DiagnosticLevel.Error))
+            {
+                throw new Exception(DiagnosticsHelper.GetDiagnosticsMessage(diagnosticsByFile));
+            }
+
+            var stringBuilder = new StringBuilder();
+            var stringWriter = new StringWriter(stringBuilder);
+
+            var emitter = new TemplateEmitter(compilation.GetEntrypointSemanticModel(), emitterSettings);
+            emitter.Emit(stringWriter);
+
+            return stringBuilder.ToString();
+        }
+
+        private Compilation GetCompilation(DocumentUri documentUri)
+        {
+            var fileUri = documentUri.ToUri();
```

Comment:
We need to get the configuration off of the documentUri. This gets used in Compilation creation and passed on to SemanticModel. We cannot avoid this as we need the diagnostics info.

The below are not based off of documentUri:

- moduleDispatcher
- features
- namespaceProvider
- fileResolver

I am a bit hesitant to modify or change anything here as it would need changes outside of this PR. I am going to leave it as is and call it - by design. 

Let me know if I am missing something or if you want me to log a bug and we can discuss it with the team. Thanks!